nixos/modules/services/web-apps/funkwhale/funkwhale.nix

```diff
@@ -0,0 +1,329 @@
+      };
+
+      static_root = mkOption {
+        type = types.str;
+        default = "/srv/funkwhale/data/static";
+        description = ''
+          Where static files (such as API css or icons) should be compiled on your system?
+          (Ensure this directory actually exists)
+        '';
+      };
+
+      django_allowed_hosts = mkOption {
+        type = types.str;
+        default = "yourdomain";
+        description = ''
+          Update it to match the domain that will be used to reach your funkwhale instance
+        '';
+      };
+
+      django_secret_key = mkOption {
+        type = types.str;
+        default = "";
+        description = ''
+          Generate one using `openssl rand -base64 45`, for example
+        '';
+      };
+    };
+
+    music_directory_path = mkOption {
+        type = types.str;
+        default = "/srv/funkwhale/data/music";
+        description = ''
+          In-place import settings
+        '';
+      };
+
+    };
+  };
+
+  config = mkIf cfg.enable {
+    services.ntp.enable = true; 
+    services.redis.enable =  true;
+    services.postgresql = {
+      enable = true;
+      # package = pkgs.postgresql100;
+      # enableTCPIP = true;
+      authentication = pkgs.lib.mkOverride 10 ''
+        local all all trust
+        host all all ::1/128 trust
+      '';
+      initialScript = pkgs.writeText "backend-initScript" ''
+        CREATE ROLE funkwhale WITH LOGIN PASSWORD 'funkwhale' CREATEDB;
+        CREATE DATABASE funkwhale WITH ENCODING 'utf8';
+        GRANT ALL PRIVILEGES ON DATABASE funkwhale TO funkwhale;
+      '';
+    };
+    services.nginx = {
+      enable = true;
+      recommendedTlsSettings = true;
+      recommendedOptimisation = true;
+      recommendedGzipSettings = true;
+      recommendedProxySettings = true;
+      appendHttpConfig = ''
+        upstream funkwhale-api {
+          server ${cfg.api_ip}:${cfg.api_port};
+        }
+      '';
+      virtualHosts = {
+        "${cfg.hostname}" = {
+        # enableACME = true; #Ask Let's Encrypt to sign a certificate for this vhost
+        # addSSL = true;
+        # forceSSL = true;
+          root = "${pkgs.funkwhale}/front";
+          default = true;
+          locations = {
+            "/" = {
+              proxyWebsockets = true;
+              tryFiles = "$uri $uri/ @rewrites";
+            };
+            "@rewrites".extraConfig = ''
+                rewrite ^(.+)$ /index.html last;
+              '';
+            "/api/".proxyPass = "http://funkwhale-api/api/";
+            "/federation/".proxyPass = "http://funkwhale-api/federation/";
+            "/rest/".proxyPass = "http://funkwhale-api/api/subsonic/rest/";
+            "/.well-known/".proxyPass = "http://funkwhale-api/.well-known/";
+            "/media/".alias = "${cfg.api.media_root}/";
+            "/_protected/media" = {
+              extraConfig = ''
+                internal;
+              '';
+              alias = "${cfg.api.media_root}";
```

Comment:
All such options can be set directly: `alias = cfg.api.media_root;`.

Reply:
fixed